eng/scripts/CodeChecks.ps1

```diff
@@ -94,6 +96,44 @@ try {
         & $PSScriptRoot\Export-API.ps1 -ServiceDirectory $ServiceDirectory -SDKType $SDKType -SpellCheckPublicApiSurface:$SpellCheckPublicApiSurface
     }
 
+    Write-Host "Validating installation instructions"
+    Join-Path "$PSScriptRoot/../../sdk" $ServiceDirectory  `
+        | Resolve-Path `
+        | % { Get-ChildItem $_ -Filter "README.md" -Recurse } `
+        | % {
+            $readmePath = $_
+            $readmeContent = Get-Content $readmePath
+            if ($readmeContent -Match "dotnet add .*--version")
+            {
+                LogError "Specific versions should not be specified in the installation instructions in '$readmePath'. For beta versions, include the --prerelease flag."
+            }
+            
+            if ($readmeContent -Match "dotnet add")
+            {
+                $changelogPath = Join-Path $(Split-Path -Parent $readmePath) "CHANGELOG.md"
+                $changeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $changelogPath
+                $hasGa = $false;
+                foreach ($key in $changeLogEntries.Keys)
+                {
+                    $entry = $changeLogEntries[$key]
+                    if ($entry.ReleaseStatus -ne "(Unreleased)" -and $entry.ReleaseVersion -notmatch "beta")
+                    {
+                        $hasGa = $true;
+                        break
+                    }
+                }
+                if ($hasGa)
+                {
+                    if (-Not ($readmeContent -Match "dotnet add (?!.*--prerelease)")){
```

Comment:
Might be worth while to add a some text describing examples of valid and invalid installation instructions just in case it's not clear.

Reply:
Perhaps, but this check is predicated on there already being a dotnet add entry, so the only thing we are really checking is that there is no version flag, and that there is one without a prerelease flag if already GAed. If we expand this check to look for the presence of these instructions, then I would agree having an example would be helpful.